### PR TITLE
[BE2] prepare for Scala 3 migration

### PR DESCRIPTION
**Pull request by bbjubjub2494**
this is a subset of #1111 that can be applied without actually switching to Scala 3. My hope is proactively doing these changes now will reduce friction when we are ready to switch.

### CODE REVIEW

File: be2-scala/src/test/scala/ch/epfl/pop/model/network/method/message/data/socialMedia/AddChirpSuite.scala [hunk 1]
```diff
@@ -28,7 +29,7 @@ class AddChirpSuite extends FunSuite with Matchers {
 object AddChirpExample {
   final val TEXT: String = "text"
   final val PARENT_ID: Option[String] = None
-  final val TIMESTAMP = Timestamp(0)
+  final val TIMESTAMP: Timestamp = Timestamp(0)
```
Comment (rombirli) [inline be2-scala/src/test/scala/ch/epfl/pop/model/network/method/message/data/socialMedia/AddChirpSuite.scala:32]: I think at some places it was unnecessary to add the typing

File: be2-scala/src/test/scala/ch/epfl/pop/model/objects/SignatureSuite.scala [hunk 2]
```diff
@@ -4,27 +4,29 @@ import java.nio.charset.StandardCharsets
...
   }
   //Data used for testing signature
-  final val dataTest = Seq("PoP-scala", "HelloWorld", "Not true is false",
+  final val dataTest: Seq[String] = Seq("PoP-scala", "HelloWorld", "Not true is false",
     "😀", "OMEGA \u03A9", "\u03A8",
     "Non empty can be fully non empty",
     "Not false is true")
 
-  final val verify_pk = PublicKey(Base64Data.encode(tester.keyPair.getPublicKey))
+  final val verify_pk: PublicKey = PublicKey(Base64Data.encode(tester.keyPair.getPublicKey))
```
Comment (rombirli) [inline be2-scala/src/test/scala/ch/epfl/pop/model/objects/SignatureSuite.scala:29]: Here too

**Review by rombirli — APPROVED** [review]
Looks really good overall

File: be2-scala/src/test/scala/ch/epfl/pop/model/network/method/message/data/socialMedia/AddChirpSuite.scala (hunk 1 above)
Comment (lourkeur) [reply be2-scala/src/test/scala/ch/epfl/pop/model/network/method/message/data/socialMedia/AddChirpSuite.scala:32]: I personally don't mind but the "migrate" plugin made that change.

File: be2-scala/src/test/scala/ch/epfl/pop/model/objects/SignatureSuite.scala (hunk 2 above)
Comment (lourkeur) [reply be2-scala/src/test/scala/ch/epfl/pop/model/objects/SignatureSuite.scala:29]: same

**Review by tuomas-p — DISMISSED** [review]
LGTM, as this is for pure migration purposes, the coverage failure shouldn't be considered relevant imo

**Comment by lourkeur** [comment]
@jbsv @pierluca I need either of you to force-merge this PR. The coverage check is in the way.

The rationale is that the changes are automated rewrites that are functionally equivalent. They are peppered all over the place so there's no clear behavior that I should go write a test about.

**Comment by K1li4nL** [comment]
Done in #1742

**Pull request closed without merging**